app/javascript/components/GooglePayButton.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import React, { Component } from 'react';
+import googlePayment from 'braintree-web/google-payment';
+import { connect } from 'react-redux';
+
+import type { AppState, PaymentMethod } from '../state';
+
+type Props = {
+  client: ?any,
+  onSubmit: () => void,
+} & typeof mapStateToProps;
+
+type State = {
+  paymentsClient?: Object,
+  googlePaymentInstance?: Object,
+  ready: boolean,
+};
+
+// TODO:
+// Do with this as you please...
+// Do we want to display google pay only on androids? Unlike Apple Pay,
+// Google Pay not restricted to Android devices.
+const isAndroid = navigator.userAgent.toLowerCase().indexOf('android') >= 0;
+
+export class GooglePayButton extends Component {
+  props: Props;
+  state: State;
+  buttonRef: any;
+  constructor(props: Props) {
+    super(props);
+    // $FlowIgnore
+    this.buttonRef = React.createRef();
+    this.state = {
+      paymentsClient: undefined,
+      googlePaymentInstance: undefined,
+      ready: false,
+    };
+  }
+
+  createGooglePayButton = () => {
+    $.ajax({
+      url: 'https://pay.google.com/gp/p/js/pay.js',
+      dataType: 'script',
+    }).then(() => {
+      if (window.google) {
+        const paymentsClient = new window.google.payments.api.PaymentsClient({
+          environment:
+            process.env.NODE_ENV === 'production' ? 'PRODUCTION' : 'TEST',
+        });
+
+        const button = paymentsClient.createButton({
+          onClick: this.onClick,
+          buttonType: 'short',
+        });
+        this.buttonRef.current.appendChild(button);
+
+        this.setState({ paymentsClient });
+      }
+    });
+  };
+
+  setupGooglePay() {
+    const { client } = this.props;
+    const { paymentsClient } = this.state;
+    if (client && paymentsClient) {
+      googlePayment.create({ client }, (err, googlePaymentInstance) => {
+        paymentsClient
+          .isReadyToPay({
+            allowedPaymentMethods: googlePaymentInstance.createPaymentDataRequest()
+              .allowedPaymentMethods,
+          })
+          .then(response => {
+            if (response.result) {
+              this.setState({
+                paymentsClient,
+                googlePaymentInstance,
+                ready: true,
+              });
+            }
+          });
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.createGooglePayButton();
+    this.setupGooglePay();
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (
```

Comment:
When we first mount the component, the braintree client might not be initialised (depending on network latency for example). When the braintree client is initialised, it's passed down as props and the component is updated. The conditional is to avoid setting up the google pay button if neither the braintree client nor the googlepay client have changed.